src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1644,7 +1647,10 @@ public function callNamedScope($scope, array $parameters = [])
      */
     public function toArray()
     {
-        return array_merge($this->attributesToArray(), $this->relationsToArray());
+        return $this->once(
+            fn () => array_merge($this->attributesToArray(), $this->relationsToArray()),
+            $this->attributesToArray(),
+        );
```

Comment:
```suggestion
        $attributes = $this->attributesToArray();
        return $this->once(
            fn () => array_merge($attributes, $this->relationsToArray()),
            $attributes,
        );
```

This kind of methods are a bit expensive so I would avoid calling it twice

Reply:
It actually winds up only being called twice for the entire stack, no matter how many times it gets called (because the second call is what gets cached).

Reply:
I've updated the default to accept a callback, and cache that value.

Your suggested change wouldn't prevent `attributesToArray()` from being called on each recursive call to `toArray()` - it would just ignore it on subsequent calls.